autoprotocol/builders.py

```diff
@@ -1557,38 +1576,127 @@ def device_mode_params(
         Raises
         ------
         ValueError
-            If model is not None or "high_volume"
-        ValueError
-            If nozzle is not None or "standard"
-        ValueError
-            If chip_material is not None or in the allowable list of tempest chip materials
-        """
-        # tempest chip specification
-        if (model != "high_volume") and (model is not None):
-            raise ValueError(f"Tempest chip model is {model}. It must be: high_volume.")
-        if (nozzle != "standard") and (nozzle is not None):
-            raise ValueError(f"Tempest nozzle is {nozzle}. It must be: standard.")
-        if (chip_material not in ["silicone", "pfe"]) and (chip_material is not None):
-            raise ValueError(
-                f"Tempest chip material is {chip_material}. It must be either: silicone or pfe."
-            )
-
-        device_mode_params = {}
-        if any([model, chip_material, nozzle]):
-            if model is None:
-                model = "high_volume"
-            if chip_material is None:
-                chip_material = "pfe"
-            if nozzle is None:
-                nozzle = "standard"
-            x_tempest_chip = {
+            If device is not of the following: [x_tempest_chip, x_mantis]
+        """
+        params: OrderedDict = OrderedDict(
+            {
                 "model": model,
                 "material": chip_material,
                 "nozzle": nozzle,
+                "diaphragm": diaphragm,
+                "nozzle_size": nozzle_size,
+                "tubing": tubing,
+                "z_drop": z_drop,
+                "viscosity": viscosity,
+            }
+        )
+        if device not in ["x_mantis", "x_tempest_chip"]:
+            raise ValueError(
+                f"Device is {device}. It must be: [x_tempest_chip, x_mantis]"
+            )
+        if any(params.values()):
```

Comment:
what do you think about nesting lines 1600-1607 under this, too? my thinking here is like we don't need to traverse this dict if there are no parameters set?

we'd have to pull `default_dict` out i guess, so we could return it

Reply:
ahh yeah ok makes sense!